sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1377,7 +1375,16 @@ public Node visitDoubleColonCast(SqlBaseParser.DoubleColonCastContext context) {
 
     @Override
     public Node visitFromStringLiteralCast(SqlBaseParser.FromStringLiteralCastContext context) {
-        return new Cast((Expression) visit(context.stringLiteral()), (ColumnType) visit(context.dataType()));
+        ColumnType targetType = (ColumnType) visit(context.dataType());
+        if (targetType.type() != ColumnType.Type.PRIMITIVE) {
+            throw new UnsupportedOperationException("type 'string' cast notation only supports primitive types. " +
+                                                    "Use '::' or cast() operator instead.");
+        }
+        if (targetType instanceof ObjectColumnType) {
```

Comment:
wouldn't it be enough just to the date type check that we have above?

Reply:
no, for some reason the `ObjectColumnType` results in having type: `Primitive` which is a bit weird. I'll try to change this if there's no big impact.

Thanks for catching this.